tabs/osd.js

```diff
@@ -2354,15 +2365,36 @@ OSD.GUI.checkAndProcessSymbolPosition = function(pos, charCode) {
 OSD.GUI.updateVideoMode = function() {
     // video mode
     var $videoTypes = $('.video-types').empty();
-    for (var i = 0; i < OSD.constants.VIDEO_TYPES.length; i++) {
 
-        $videoTypes.append(
-            $('<label/>')
-            .append($('<input name="video_system" type="radio"/>' + OSD.constants.VIDEO_TYPES[i] + '</label>')
-                .prop('checked', i === OSD.data.preferences.video_system)
-                .data('type', i)
-            )
-        );
+    if (!OSD.data.isDjiHdFpv) {
+        $('#dji_settings').hide();
+    }
+
+    if (OSD.data.isMspDisplay) {
+        if (OSD.constants.VIDEO_TYPES[OSD.data.preferences.video_system] != 'HDZERO') {
+            OSD.data.preferences.video_system = OSD.constants.VIDEO_TYPES.indexOf('HDZERO');
+            OSD.updateDisplaySize();
+            OSD.GUI.saveConfig();
+        }
+    } else {
+        if (OSD.constants.VIDEO_TYPES[OSD.data.preferences.video_system] == 'HDZERO' || OSD.constants.VIDEO_TYPES[OSD.data.preferences.video_system] == 'DJIWTF') {
+            OSD.data.preferences.video_system = OSD.constants.VIDEO_TYPES.indexOf('AUTO')
+            OSD.updateDisplaySize();
+            OSD.GUI.saveConfig();
+        }
+    }
+
+    for (var i = 0; i < OSD.constants.VIDEO_TYPES.length; i++) {
+        if ((OSD.constants.VIDEO_TYPES[i] != 'HDZERO' && OSD.constants.VIDEO_TYPES[i] != 'DJIWTF') || OSD.data.isMspDisplay)
```

Comment:
There is a bug that if you have DJIWTF selected and saved. If you leave the OSD page and return, HDZERO is selected instead. Is should show that you are using DJIWTF. I think the bug is in the above IF statement. I think it should be something like:

```
if ( OSD.data.isMspDisplay || 
     (!OSD.data.isMspDisplay && (OSD.constants.VIDEO_TYPES[i] === 'HDZERO' || OSD.constants.VIDEO_TYPES[i] === 'DJIWTF'))
   )
```

Reply:
I was working, then I kept adding changes. Must have broke it in the process. Will fix.